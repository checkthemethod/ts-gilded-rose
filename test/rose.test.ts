import GildedRose, { Item } from '../src';

/* TODO: move coverage tests to own individual tests */
const initializeData = () => {
  const items = [];
  items.push(new Item('+5 Dexterity Vest', 10, 20));
  items.push(new Item('Aged Brie', 2, 0));

  items.push(new Item('Elixir of the Mongoose', 5, 7));
  items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
  // add coverage test for SULFURAS no change in quality
  items.push(new Item('Sulfuras, Hand of Ragnaros', -1, 80));
  items.push(
    new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20)
  );
  // add coverage test for sellin < 11
  items.push(
    new Item('Backstage passes to a TAFKAL80ETC concert', 10, 49)
  );
  // add coverage test for sellin < 6
  items.push(
    new Item('Backstage passes to a TAFKAL80ETC concert', 5, 45)
  );
  // add coverage test for quality increase when sellIn < 0
  items.push(
    new Item('Aged Brie', -1, 49)
  );
  items.push(new Item('Conjured Mana Cake', 3, 6));
  return new GildedRose(items)
}

const deepClone = (data: Item[]) => {
  let cloneItems = JSON.parse(JSON.stringify(data))
  return cloneItems;
}

describe('default update test', () => {
  it('works', () => {
    expect(new GildedRose()).not.toBeNull();
  });
});

describe('update with passed items', () => {
  const gildedRose = initializeData()
  it('works', () => {
    expect(gildedRose).not.toBeNull();
  });
});

describe('updateQuality Tests', () => {
  it('updateQuality data matches snapshot after x iterations, default: 15', () => {
    const gildedRose = initializeData();
    var numIterations = 20;
    var expected = [];
    for(var i=0; i< numIterations;i++) {
      expected.push(deepClone(gildedRose.items));
      gildedRose.updateQuality();
    }
    expect(expected).toMatchSnapshot()
  })
})
