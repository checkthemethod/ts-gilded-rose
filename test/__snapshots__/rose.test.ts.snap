// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateQuality Tests updateQuality data matches snapshot after x iterations, default: 15 1`] = `
Array [
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 45,
      "sellIn": 5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": -1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 19,
      "sellIn": 9,
    },
    Object {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 1,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 6,
      "sellIn": 4,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 21,
      "sellIn": 14,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 48,
      "sellIn": 4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 4,
      "sellIn": 2,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 18,
      "sellIn": 8,
    },
    Object {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 0,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 5,
      "sellIn": 3,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 22,
      "sellIn": 13,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 2,
      "sellIn": 1,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 17,
      "sellIn": 7,
    },
    Object {
      "name": "Aged Brie",
      "quality": 4,
      "sellIn": -1,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 4,
      "sellIn": 2,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 23,
      "sellIn": 12,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -4,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": 0,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 16,
      "sellIn": 6,
    },
    Object {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": -2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 3,
      "sellIn": 1,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 24,
      "sellIn": 11,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 6,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 15,
      "sellIn": 5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 8,
      "sellIn": -3,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 2,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 25,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 0,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -6,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 14,
      "sellIn": 4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 10,
      "sellIn": -4,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -1,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 27,
      "sellIn": 9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -7,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 13,
      "sellIn": 3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 12,
      "sellIn": -5,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 29,
      "sellIn": 8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -8,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -4,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 12,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 14,
      "sellIn": -6,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -3,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 31,
      "sellIn": 7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -9,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -5,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 11,
      "sellIn": 1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": -7,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -4,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 33,
      "sellIn": 6,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -10,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -6,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 10,
      "sellIn": 0,
    },
    Object {
      "name": "Aged Brie",
      "quality": 18,
      "sellIn": -8,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 35,
      "sellIn": 5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 0,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -11,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -7,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 8,
      "sellIn": -1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 20,
      "sellIn": -9,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -6,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 38,
      "sellIn": 4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -6,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -12,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -8,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 6,
      "sellIn": -2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 22,
      "sellIn": -10,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -7,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 41,
      "sellIn": 3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -7,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -13,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -9,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 4,
      "sellIn": -3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 24,
      "sellIn": -11,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -8,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 44,
      "sellIn": 2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -8,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -14,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -10,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 2,
      "sellIn": -4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 26,
      "sellIn": -12,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -9,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 47,
      "sellIn": 1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -9,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -15,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -11,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 0,
      "sellIn": -5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 28,
      "sellIn": -13,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -10,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 0,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -16,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -12,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 0,
      "sellIn": -6,
    },
    Object {
      "name": "Aged Brie",
      "quality": 30,
      "sellIn": -14,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -11,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -6,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -11,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -17,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -13,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 0,
      "sellIn": -7,
    },
    Object {
      "name": "Aged Brie",
      "quality": 32,
      "sellIn": -15,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -12,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -12,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -18,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -14,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 0,
      "sellIn": -8,
    },
    Object {
      "name": "Aged Brie",
      "quality": 34,
      "sellIn": -16,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -13,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -13,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -19,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -15,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 0,
      "sellIn": -9,
    },
    Object {
      "name": "Aged Brie",
      "quality": 36,
      "sellIn": -17,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -14,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -14,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -20,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -16,
    },
  ],
]
`;
